rpadmin/api_migration.go

```diff
@@ -53,16 +61,22 @@ func (a *AdminAPI) DeleteMigration(ctx context.Context, id int) error {
 	return a.sendAny(ctx, http.MethodDelete, fmt.Sprintf("baseMigrationEndpoint%d", id), nil, nil)
 }
 
-// ExecuteMigration executes a specific action on a migration identified by its ID. The action must be one of:
-// prepare, execute, finish, cancel.
-func (a *AdminAPI) ExecuteMigration(ctx context.Context, id int, action string) error {
-	validActions := map[string]bool{
-		"prepare": true,
-		"execute": true,
-		"finish":  true,
-		"cancel":  true,
-	}
-	if !validActions[action] {
+type MigrationAction int
+
+const (
+	PrepareAction MigrationAction = iota
+	ExecuteAction
+	FinishAction
+	CancelAction
+)
+
+func (a MigrationAction) String() string {
+	return [...]string{"prepare", "execute", "finish", "cancel"}[a]
+}
```

Comment:
I know this saves a lot of lines, but I do prefer to have explicit returns here, with a simple type-switch to improve readability and handle out-of-range cases

```
switch a {
  case PrepareAction:
    return "prepare"
  ...
  default:
    return ""
}
```

Also, this can result in a panic due to index out of range if the caller creates a migration manually:
```
 a := MigrationAction(1111)
 a.String() // Panic
```
https://go.dev/play/p/z24KxsdzooR